tests/test_seq2pat_batch.py

```diff
@@ -0,0 +1,284 @@
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: GPL-2.0
+import collections
+import os
+import unittest
+
+from sequential.seq2pat import Seq2Pat, Attribute
+from sequential.utils import read_data, sort_pattern, list_to_counter, counter_to_list
+
+
+class TestSeq2PatBatch(unittest.TestCase):
+    TEST_DIR = os.path.dirname(os.path.abspath(__file__))
+    DATA_DIR = TEST_DIR + os.sep + "data" + os.sep
+
+    def test_list_to_counter(self):
+        patterns = [[11, 12, 1],
+                    [11, 12, 13, 1],
+                    [11, 13, 1],
+                    [12, 13, 1]]
+
+        counter = list_to_counter(patterns)
+
+        self.assertEqual(counter, collections.Counter({(11, 12): 1, (11, 12, 13): 1, (11, 13): 1, (12, 13): 1}))
+
+    def test_counter_to_list(self):
+        counter = collections.Counter({(11, 12): 1, (11, 12, 13): 2, (11, 13): 1, (12, 13): 1})
+
+        results = counter_to_list(counter, min_row_count=2)
+
+        self.assertEqual(results, [[11, 12, 13, 2]])
+
+    def test_batch_patterns_default_equal(self):
+        # Seq2Pat over 3 sequences
+        seq2pat = Seq2Pat(sequences=[["A", "C", "B", "A", "D"],
+                                     ["C", "B", "A"],
+                                     ["C", "A", "C", "D"]])
+
+        # Patterns
+        patterns = seq2pat.get_patterns(min_frequency=1.0)
+
+        seq2pat = Seq2Pat(sequences=[["A", "C", "B", "A", "D"],
+                                     ["C", "B", "A"],
+                                     ["C", "A", "C", "D"]], batch_size=2)
```

Comment:
so how does this work? It uses first 2 rows as batch 1 and the remaining 1 row as batch 2? IF there were 6 rows, we would do batches of 2 x 3? And if there were 7 rows, we would a batch of 3 + 4?

Reply:
So if 6 rows, I think yes, it will go 2 x 3. Then if 7 rows, it will go 2 x 3, then 1 x 1, making it 4 batches.